src/main/java/com/example/demo/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,22 @@
+package com.example.demo.security;
+
+import com.example.demo.exception.EntityNotFoundException;
+import com.example.demo.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String email)
+            throws UsernameNotFoundException {
+        return (UserDetails) userRepository.findByEmail(email)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find user by email"));
```

Comment:
```suggestion
                .orElseThrow(() -> new UsernameNotFoundException("Can't find user by email"));
```
Also, add email parameter